software/firmware/europi.py

```diff
@@ -59,6 +59,18 @@
 CHAR_HEIGHT = 8
 
 
+
+# EuroPi error class
+
+class EuroPiError(Exception):
+    def __init__(self, error_code):
+        oled.fill(0)
+        oled.centre_text(f'Error {error_code}')
+        oled.show()
+        time.sleep(5)
+        oled.fill(0)

```

Comment:
[discussion] What is the reason for hiding the error message? I think we would want to leave it on the screen for users to take note of. If it disappears before the user notices or takes note of the error message then it would be more difficult to troubleshoot. 
```suggestion
```

Reply:
The reasoning behind this is that sometimes the error is not one that causes the module to be entirely unusable, so you might want the script to continue as normal, in which case you could get some really weird effects if you are using .scroll() or anything and the error message isn't wiped.

Maybe it should just be decided that the error should prevent the script from continuing and exit.
I think this might need some extra code for the europi script class though, because using exit() didn't work for me and the script would continue even if europi.py called exit()

Reply:
Yeah, I would argue that Exceptions are for exceptional conditions, and we shouldn't be raising them if they are not. And we definitely shouldn't be taking over the screen for a recoverable exception.

Also, the exception shouldn't need to call exit. If the exception is not caught, it should rise up the call stack and kill the running program.